accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,282 @@
+
+
+### Component 1: Exporter
+
+Goal: Configure the Exporter to gather metrics from the Uyuni server and managed systems. The exporter uses Salt runners to gather metrics.
+
+Steps:
+    Deploy the Exporter as a containerized application.
+    Ensure the exporter is accessible for Prometheus to scrape metrics.
+    Ensure the exporter can run Salt runner commands targeting the server.
+
+### Component 2: Prometheus
+
+Goal: Set up Prometheus to scrape metrics from the Uyuni-Health-Check exporter and evaluate alerting rules.
+
+Steps:
+    Scrape Configuration: Add the Exporter endpoint to the Prometheus configuration.
+    Alerting Rules Definition: Create alerting rules to define conditions for triggering alerts based on metrics from the Exporter.
+    Alertmanager Configuration: Set up Alertmanager to handle alerts generated by Prometheus, including notifications.
+
+### Component 3: Grafana
+
+Goal: Use Grafana to visualize metrics and alerts from Prometheus.
+
+Steps:
+    Add Prometheus as a data source in Grafana.
+    Create dashboards to visualize metrics from the Exporter.
+    Configure panels within dashboards to display alerts based on Prometheus data.
+
+### Component 6: Loki and Promtail
+
+**Loki setup**
+Goal: Deploy and configure Loki to serve as the centralized log aggregation system for collecting, storing, and querying logs from the Uyuni environment.
+
+Steps:
+    Deployment: Install Loki on the Uyuni infrastructure as a container that is part of the Uyuni-Health-Check Pod.
+    Configuration: Customize the Loki configuration to define storage locations for logs, retention policies, and other operational parameters.
+    Service Discovery: Configure Loki to discover targets for log collection, focusing on integration with the Exporter and Uyuni components.
+
+**Promtail setup**
+Goal: Configure Promtail to collect logs from Uyuni server and managed systems, forwarding them to Loki for aggregation and analysis.
+
+Steps:
+    Deployment: Install Promtail on the Uyuni infrastructure as a container that is part of the Uyuni-Health-Check Pod.
+    Configuration:
+        Edit Promtail configuration to define the paths of log files to monitor. In the containerised Uyuni environment, the Promtail container must be configured to access mapped volumes that correspond to the log locations of the Uyuni server container.
+        Configure Promtail to forward logs to your Loki instance.
+
+### Component 7: LogCLI
+Goal: Incorporate LogCLI to provide a command-line interface for querying logs stored in Loki, enhancing the toolset available for diagnostics, troubleshooting, and operational reporting.
```

Comment:
LogCLI will allow the user to query from all aggregated logs via CLI. This would allow to easy query the logs without going to a UI.

Agree that this is an extra component to maintain that it is maybe not bringing that many benefits if we already provide a Grafana dashboard to visualize the data.

Reply:
I'm not certain about its feasibility, but it would still be beneficial to have it available for engineers and support teams, even if we don't include it in the customer shipment.